legend-engine-xt-serviceStore-executionPlan/src/main/java/org/finos/legend/engine/plan/execution/stores/service/auth/ServiceStoreConnectionProvider.java

```diff
@@ -0,0 +1,131 @@
+import org.finos.legend.engine.connection.ConnectionSpecification;
+import org.finos.legend.engine.plan.execution.stores.service.IServiceStoreExecutionExtension;
+import org.finos.legend.engine.protocol.pure.v1.model.executionPlan.nodes.AuthenticationSchemeRequirement;
+import org.finos.legend.engine.protocol.pure.v1.model.executionPlan.nodes.SingleAuthenticationSchemeRequirement;
+import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.authentication.specification.AuthenticationSpecification;
+import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.store.service.model.SecurityScheme;
+import org.finos.legend.engine.shared.core.function.Function5;
+import org.finos.legend.engine.shared.core.identity.Credential;
+import org.finos.legend.engine.shared.core.identity.Identity;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+public class ServiceStoreConnectionProvider extends ConnectionProvider<HttpConnectionBuilder>
+{
+
+    public ServiceStoreConnectionProvider(CredentialProviderProvider credentialProviderProvider)
+    {
+        super(credentialProviderProvider);
+    }
+
+    public HttpConnectionBuilder makeConnection(ConnectionSpecification connectionSpec, AuthenticationSpecification authenticationSpec, Identity identity) throws Exception
+    {
+        assert (connectionSpec instanceof ServiceStoreConnectionSpecification);
+        ServiceStoreConnectionSpecification serviceStoreConnectionSpecification = (ServiceStoreConnectionSpecification) connectionSpec;
+
+        assert (authenticationSpec instanceof ServiceStoreAuthenticationSpecification);
+        ServiceStoreAuthenticationSpecification serviceStoreAuthenticationSpecification = (ServiceStoreAuthenticationSpecification) authenticationSpec;
+
+        for (AuthenticationSchemeRequirement requirement : serviceStoreAuthenticationSpecification.authenticationSchemeRequirements)
+        {
+            SingleAuthenticationSchemeRequirement authenticationSchemeRequirement = (SingleAuthenticationSchemeRequirement) requirement;
+            SecurityScheme securityScheme = authenticationSchemeRequirement.securityScheme;
+            AuthenticationSpecification authenticationSpecification = authenticationSchemeRequirement.authenticationSpecification;
+
+            try
+            {
+                HttpClientBuilder clientBuilder = HttpClients.custom();
+                RequestBuilder requestBuilder = makeRequestUtil(serviceStoreConnectionSpecification);
+                validateSecurityScheme(securityScheme, authenticationSpecification);
+                Credential credential = processSecurityScheme(securityScheme, authenticationSpecification, identity);
+                if (credential != null)
+                {
+                    configureAuthentication(requestBuilder, clientBuilder, securityScheme, authenticationSpecification, credential);
```

Comment:
why we need to pass authScheme here, we have already process auth to get credential why do we need it in processor?